### ISSUE TITLE
no-unsafe-mutable-readonly-assignment: doesn't catch call signature assignment when params are omitted

### ISSUE THREAD

**Issue by danielnixon** [danielnixon/eslint-plugin-total-functions#736]
```ts
interface A {
  (a: { readonly a: string }, s: string): number;
}

interface B {
  // eslint-disable-next-line functional/prefer-immutable-types
  (a: { a: string }, s: string): number;
}

declare const a: A;

// Flagged by total-functions/no-unsafe-mutable-readonly-assignment
const b: B = a;
```

but:

```ts
interface A {
  (a: { readonly a: string }, s: string): number;
}

interface B {
  // eslint-disable-next-line functional/prefer-immutable-types
  (a: { a: string }): number;
}

declare const a: A;

// Not flagged
const b: B = a;
```

**Comment by danielnixon** [comment]
https://www.typescriptlang.org/docs/handbook/type-compatibility.html#comparing-two-functions

**Comment by danielnixon** [comment]
🤔 
```ts
let identity = function <T>(x: T): T {
  return x;
};

let makeReadonly = function <U>(y: U): Readonly<U> {
  return y;
};

// Not flagged
identity = makeReadonly;

// Not flagged
makeReadonly = identity;
```
